### PR TITLE
Sai ptf server v2 user guide

### PR DESCRIPTION
**Pull request by reshmaintel**
User Guide for SAI thrift server.  Adds info on how a user can generate automated test framework and integrate it with the user libsai.

### CODE REVIEW

**Comment by chrispsommers** [comment]
LGTM, I was able to execute these instructions. If https://github.com/opencomputeproject/SAI/pull/1514 gets merged, the instructions should be updated to show how to pass option flags. If this gets merged first, I can add that to https://github.com/opencomputeproject/SAI/pull/1514as well (i.e. update this README).

**Comment by chrispsommers** [comment]
Hi @reshmaintel you need to rebase and signoff for the DCO to pass, instructions for this are here: https://github.com/opencomputeproject/SAI/pull/1512/checks?check_run_id=7046496912 (I did same on my https://github.com/opencomputeproject/SAI/pull/1514 and had to fix, oops)

**Comment by chrispsommers** [comment]
@vmytnykx I will update the User guide after https://github.com/opencomputeproject/SAI/pull/1514 is merged.

File: ptf/SAI_PTF_user-guide.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+```bash
+apt update
+apt install git python python-pip wget doxygen graphviz aspell
+apt install libgetopt-long-descriptive-perl libconst-fast-perl \
+libtemplate-perl libnamespace-autoclean-perl libmoose-perl libmoosex-aliases-perl
+pip install ctypesgen
+```
+
+Apache thrift 0.11.0
```
Comment (richardyu-ms) [inline ptf/SAI_PTF_user-guide.md:27]: https://github.com/Azure/sonic-buildimage/pull/9884

it should be able to build it with thrift 0.14.0

**Review by richardyu-ms — COMMENTED** [review]
Besides the thrift version, others are LGTM. Thanks for this useful doc!

File: ptf/SAI_PTF_user-guide.md (hunk 1 above)
Comment (vmytnykx) [reply ptf/SAI_PTF_user-guide.md:27]: @richardyu-ms  yes, the saithrift can be built with thrift 0.14.1 also. Verified already. Once this PR is merged, I will update the doc with thrift 0.14 info included.

**Pull request merged**